### PR TITLE
Module tests

### PR DESCRIPTION
**Pull request by gauntface**
Small fix to stop obscure input and opening and leaving a tab open.

### CODE REVIEW

File: test/module/index.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      connection.destroy();
+    });
+    testServer.close(cb);
+  });
+
+  it('should have a main defined in the package.json', function() {
+    assert.ok(pkg.main);
+  });
+
+  it('should be able to require in the pacakge.json\'s main file', function() {
```
Comment (paullewis) [inline test/module/index.js]: tyop

**Comment by paullewis** [comment]
Nit; otherwise LGTM!

**Comment by gauntface** [comment]
@paullewis fixed the typo

File: test/module/index.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/* eslint-env mocha */
```
Comment (brendankenny) [inline test/module/index.js]: fancy

Comment (brendankenny) [inline test/module/index.js]: (we should be doing this in all our other tests :)

File: test/module/index.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+  it('should throw an error when the first parameter is not defined', function(cb) {
+    const lighthouse = require(path.join('..', '..', pkg.main));
+    return lighthouse()
+    .then(() => {
+      assert.error(new Error('Should not have resolved when first arg is not a string'));
+    })
+    .catch(err => {
+      assert.ok(err);
+    })
+    .then(cb);
```
Comment (brendankenny) [inline test/module/index.js]: since you're returning the promise (and handling any errors), no need for `cb`?

Comment (brendankenny) [inline test/module/index.js]: (same for below)

File: test/module/index.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    });
+    testServer.close(cb);
+  });
+
+  it('should have a main attribute defined in the package.json', function() {
+    assert.ok(pkg.main);
+  });
+
+  it('should be able to require in the package.json\'s main file', function() {
+    const lighthouse = require(path.join('..', '..', pkg.main));
```
Comment (brendankenny) [inline test/module/index.js]: nit: `path.join` on these seems overly defensive, slashes work fine

**Comment by brendankenny** [comment]
> leaving a tab open

did this refer to something that used to be in the PR?

**Comment by brendankenny** [comment]
looking good with some minor feedback.

This can be a replacement for `smoke-test.sh`, though we'll also want a test for a page that actually engages with some PWA features for that. We need to watch the number of runs we do (this runs two), though. Lighthouse already takes a long time and it's only getting longer :)

File: test/module/index.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    assert.ok(lighthouse);
+  });
+
+  it('should require lighthouse as a function', function() {
+    const lighthouse = require(path.join('..', '..', pkg.main));
```
Comment (brendankenny) [inline test/module/index.js]: totally nitty nit: since `pkg.main` is established as good to go, can just `const lighthouse = require('../../');` here

**Comment by paulirish** [comment]
nice stuff here. :) the env-mocha is a quality protip. 👲

File: test/module/index.js (hunk 4 above)
Comment (gauntface) [inline test/module/index.js]: Yeah, but should you ever want to test on windows it'll be useful

File: test/module/index.js (hunk 5 above)
Comment (gauntface) [inline test/module/index.js]: Oohhhh will require look for the package.json and use main from it?

File: test/module/index.js (hunk 4 above)
Comment (brendankenny) [inline test/module/index.js]: should be good. For a path like this either slash should work on windows, and I believe `require` normalizes things anyway

**Comment by gauntface** [comment]
The "leaving a tab open" comment is related to the changes to checking the url is a string with a length > 0.

If it wasn't it would make it's way to the scheduler after the tab had been opened and then error, leaving a lingering tab open.

**Comment by gauntface** [comment]
All feedback should be in now (thanks a lot).

I'm planning on taking a look at smoke-test.sh and just a little bit of test tidy up in a seperate PR if that's cool with everyone.

Looking for the +2 ;)

**Comment by paullewis** [comment]
I'm very much down with smoke-test.sh being upgraded. It failed for me earlier and left a web server hanging on :9999 which was unfortch.

**Pull request merged**